### PR TITLE
Datatable ordering fix

### LINKED ISSUE #79: Datatable Ordering fix
image

add datatable ordering feature and Use Eloquent query to order in MySQL not in array

_Originally posted by @Raza9798 in https://github.com/aanish27/employee-profile-manager/issues/75#issuecomment-2427582092_

### PR DESCRIPTION
**Pull request by aanish27**
Resolved #79 @Raza9798 Please reveiw

### CODE REVIEW

File: app/Http/Controllers/EmployeeController.php [hunk 1]
```diff
@@ -29,8 +29,10 @@ public function draw(Request $request){
         $start = $request->query('start', 0);
         $length = $request->query('length', 10);
         $totalEmployees =   Employee::count();
-        $arrayColumns = ['','company.name', 'company.branch', 'name', 'position', 'dob', 'email', 'phone', 'address', 'bankAccount.account_no'];
+        $arrayColumns = ['', 'company_name', 'branch', 'name', 'position', 'dob', 'email', 'phone', 'address', 'account_no'];
```
Comment (Raza9798) [inline app/Http/Controllers/EmployeeController.php:32]: remove this array and receive the column from client side. no point of defining an array

**Review by Raza9798 — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/EmployeeController.php (hunk 1 above)
Comment (aanish27) [reply app/Http/Controllers/EmployeeController.php]: Yes got it....made the change @Raza9798 #83

File: app/Http/Controllers/CompanyController.php [hunk 2]
```diff
@@ -34,7 +33,7 @@ public function draw(Request $request){
         }
 
         if(!empty($arr)){
-            $searchCountry = empty($arr) ? $search : (array_key_exists(3, $arr) ? $arr[3] : null);
+            $searchCountry = array_key_exists(3, $arr) ? $arr[3] : null;
```
Comment (Raza9798) [inline app/Http/Controllers/CompanyController.php:36]: use the index from parameter which sends from datatable. batter not to hardcode here. i am not able to guess key 3 represents what

**Review by Raza9798 — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/CompanyController.php (hunk 2 above)
Comment (aanish27) [reply app/Http/Controllers/CompanyController.php]: Yeah okay i refactored it using column names now the code is more readable please review again @Raza9798  #79

**Review by Raza9798 — APPROVED** [review]

**Pull request merged**